api/envoy/type/matcher/v3/string.proto

```diff
@@ -61,6 +62,11 @@ message StringMatcher {
     //
     // * ``abc`` matches the value ``xyz.abc.def``
     string contains = 7 [(validate.rules).string = {min_len: 1}];
+
+    // Use an extension as the matcher type.
+    // [#extension-category: envoy.matching.string]
+    // [#comment: This uses the xds type instead of the envoy type to avoid a circular dependency in the build system.]
+    xds.core.v3.TypedExtensionConfig extension = 8;
```

Comment:
@ggreenway what's the main difference between this and using the `safe_regex` field with a custom matcher there?

Reply:
Answering my own extension, we only have `oneof`, not pluggable extensions for `RegexMatcher`.